runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -275,27 +292,132 @@ public void onGameStateChanged(GameStateChanged event)
 				taggedNpcs.clear();
 				break;
 			case LOGGED_IN:
-				if (config.amount() != -1
-					&& !config.taskName().isEmpty()
+				if (getStoredAmount() != -1
+					&& !getStoredTaskName().isEmpty()
 					&& loginFlag)
 				{
-					setExpeditiousChargeCount(config.expeditious());
-					setSlaughterChargeCount(config.slaughter());
-					setTask(config.taskName(), config.amount(), config.initialAmount(), config.taskLocation(), false);
+					setExpeditiousChargeCount(getStoredExpeditiousCharges());
+					setSlaughterChargeCount(getStoredSlaughterCharges());
+					setTask(getStoredTaskName(),
+						getStoredAmount(),
+						getStoredInitialAmount(),
+						getStoredTaskLocation(), false);
 					loginFlag = false;
 				}
 				break;
 		}
 	}
 
+	public int getStoredAmount()
+	{
+		return getIntConfig(AMOUNT_KEY);
+	}
+
+	public void setStoredAmount(int amount)
+	{
+		setIntConfig(AMOUNT_KEY, amount);
+	}
+
+	public int getStoredInitialAmount()
+	{
+		return getIntConfig(INIT_AMOUNT_KEY);
+	}
+
+	public void setStoredInitialAmount(int initialAmount)
+	{
+		setIntConfig(INIT_AMOUNT_KEY, initialAmount);
+	}
+
+	public int getStoredPoints()
+	{
+		return getIntConfig(POINTS_KEY);
+	}
+
+	public void setStoredPoints(int points)
+	{
+		setIntConfig(POINTS_KEY, points);
+	}
+
+	public int getStoredStreak()
+	{
+		return getIntConfig(STREAK_KEY);
+	}
+
+	public void setStoredStreak(int streak)
+	{
+		setIntConfig(STREAK_KEY, streak);
+	}
+
+	public String getStoredTaskName()
+	{
+		return getStringConfig(TASK_NAME_KEY);
+	}
+
+	public void setStoredTaskName(String taskName)
+	{
+		setStringConfig(TASK_NAME_KEY, taskName);
+	}
+
+	public String getStoredTaskLocation()
+	{
+		return getStringConfig(TASK_LOC_KEY);
+	}
+
+	public void setStoredTaskLocation(String taskLocation)
+	{
+		setStringConfig(TASK_LOC_KEY, taskLocation);
+	}
+
+	public int getStoredExpeditiousCharges()
+	{
+		return getIntConfig(EXPEDITIOUS_CHARGES_KEY);
+	}
+
+	public void setStoredExpeditiousCharges(int expeditiousCharges)
+	{
+		setIntConfig(EXPEDITIOUS_CHARGES_KEY, expeditiousCharges);
+	}
+
+	public int getStoredSlaughterCharges()
+	{
+		return getIntConfig(SLAUGHTER_CHARGES_KEY);
+	}
+
+	public void setStoredSlaughterCharges(int slaughterCharges)
+	{
+		setIntConfig(SLAUGHTER_CHARGES_KEY, slaughterCharges);
+	}
+
+	private int getIntConfig(String key)
+	{
+		Integer value = configManager.getRSProfileConfiguration(SlayerConfig.GROUP_NAME, key, int.class);
+		return value == null ? -1 : value;
+	}
+
+	private void setIntConfig(String key, int value)
```

Comment:
This is identical to `setStringConfig` except for the signature. Change the signature to `(String key, Object value)` and direct all setters to use it.